metaserver/src/main/scala/scala/meta/languageserver/compiler/ScalacProvider.scala

```diff
@@ -16,9 +15,7 @@ import org.langmeta.inputs.Input
 import org.langmeta.io.AbsolutePath
 
 /** Responsible for keeping fresh scalac global instances. */
-class ScalacProvider(
-    serverConfig: ServerConfig
-) extends LazyLogging {
+object ScalacProvider extends LazyLogging {
```

Comment:
I'm not a fan of keeping state in objects, I prefer to `new X` to handle the lifecycle instead of leaving it to the JVM.

Reply:
Sure, I can make this a class with no parameters. It's probably a good idea since I suspect we may need to configure this provider too (e.g. with the version of Scala)

Reply:
Good catch @olafurpg I missed that but agree, especially from a testing standpoint.

Reply:
done!